Jenkinsfile

```diff
@@ -332,6 +348,35 @@ pipeline {
                 checkout scm
             }
         }
+        stage('check_for_changed_files') {
+          steps {
+            script {
+	      if (env.BRANCH_NAME == 'master') {
+	        return
+	      }
+
+	      def all_files = new HashSet<String>()
+
+              for (set in currentBuild.changeSets) {
```

Comment:
This is checking for changes since the last build, which is… probably correct. Might want a comment though.

(That is: if no relevant files changed since the last build on the same branch happened, this wouldn't build)

Reply:
Handled with commit e3ecd5fe538df3862b54e194b822fdd4f27cd987